packages/blocks/google-cloud/src/lib/actions/get-recommendations-cli-action.ts

```diff
@@ -0,0 +1,298 @@
+        command: currentCommand,
+        error,
+      });
+    }
+  },
+});
+
+async function getScopeOptions(auth: any, useHostSession: boolean) {
+  const dropdowns = {
+    billingAccount: createDropdown(
+      'Billing Account',
+      'The Google Cloud Platform billing account ID to use for this invocation.',
+      true,
+    ),
+    organization: createDropdown(
+      'Organization ID',
+      'The Google Cloud Platform organization ID to use for this invocation.',
+      true,
+    ),
+    project: createDropdown(
+      'Project ID',
+      'The Google Cloud Platform project ID.',
+      true,
+    ),
+  };
+
+  const scopeConfigs = [
+    {
+      key: 'billingAccount',
+      command: 'gcloud billing accounts list',
+      extractLabelAndValue: (item: { name: string; displayName: string }) => ({
+        label: item.displayName,
+        value: item.name.split('/')[1],
+      }),
+    },
+    {
+      key: 'organization',
+      command: 'gcloud organizations list',
+      extractLabelAndValue: (item: { name: string; displayName: string }) => ({
+        label: item.displayName,
+        value: item.name.split('/')[1],
+      }),
+    },
+    {
+      key: 'project',
+      command: 'gcloud projects list',
+      extractLabelAndValue: (item: { name: string; projectId: string }) => ({
+        label: item.name,
+        value: item.projectId,
+      }),
+    },
+  ];
+
+  const commands = scopeConfigs.map(
+    ({ command }) => `${command} --format=json`,
+  );
+
+  let results: string[] = [];
+  try {
+    results = await runCommands(commands, auth, useHostSession);
+  } catch (error) {
+    for (const { key } of scopeConfigs) {
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: true,
+        options: [],
+        placeholder: `Error fetching ${key}s`,
+        error: `${error}`,
+      };
+    }
+    return getScopeOptionsReturnStructure(dropdowns);
+  }
+
+  scopeConfigs.forEach(({ key, extractLabelAndValue }, index) => {
+    try {
+      const parsedItems = JSON.parse(results[index] ?? '[]');
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: false,
+        options: parsedItems.map(extractLabelAndValue),
+      };
+    } catch (error) {
+      dropdowns[key as keyof typeof dropdowns].options = {
+        disabled: true,
+        options: [],
+        placeholder: `Error parsing ${key} results`,
+        error: `${error}`,
+      };
+    }
+  });
+
+  return getScopeOptionsReturnStructure(dropdowns);
+}
+
+function getScopeOptionsReturnStructure(dropdowns: Record<string, any>) {
+  return [
+    {
+      label: 'Filter by Billing Account',
+      value: { billingAccount: dropdowns['billingAccount'] },
+    },
+    {
+      label: 'Filter by Organization ID',
+      value: { organization: dropdowns['organization'] },
+    },
+    { label: 'Filter by Project ID', value: { project: dropdowns['project'] } },
+    {
+      label: 'Filter by Folder ID',
```

Comment:
This has to be a text as the input for the cli call needs to contain either Org id, or folder id, so you can either get all folder within an orgs are below another folder